### DISCUSSION TITLE
.NET: Clearing long running tasks after a hang query

### DISCUSSION THREAD

**Discussion by itsUndefined** [ravendb/ravendb#19632] [Q&A] [answered]
After a transient loss of connection between an API and RavenDB I received the following unhandled exceptions 12 hours later:

```
The request was canceled due to the configured HttpClient.Timeout of 43200 seconds elapsing.
```

It appears that there is a `GlobalHttpClientTimeout`  that, I think, is not configurable inside the `Raven.Client` and that it's set to 12 hours.
What should I do to handle this case in the future? Is there a way to lower this Timeout to a more sensible value? Should I pass the CancellationToken from the .NET controller action into all my queries and loads so that the request is cancelled if the user and/or the ASP.NET server aborts the connection?

Edit: I am using RavenDB 5.4

**Comment by ayende** [comment, marked as answer]
That is in the Conventions, yes. You can set it to whatever value you like. 
Or pass a cancellation token, yes.

**Reply by itsUndefined** [reply]
How can I configure the GlobalHttpClientTimeout inside the Conventions? It looks like a private field to me.

**Reply by ppekrol** [reply]
in 6.0+ Client APIs it is available. See here: https://github.com/ravendb/ravendb/blob/83bff18bee184564bb733ff5eb3116fba5039eab/src/Raven.Client/Documents/Conventions/DocumentConventions.cs#L564

**Reply by georgiosd** [reply]
Could someone please elaborate on why this default is set to 12 hours? I'd love to understand the logic behind this choice before changing it.

**Reply by ppekrol** [reply]
This is the maximum request time of the any http request, not matter if this is a simple load, query or a long running patching/export/etc. All of those will be canceled if this number if exceeded, even if you are in a middle of processing. We are currently working on (7.x change) lowering this value and providing different mechanisms for long running operations.
